kytos/core/auth.py

```diff
@@ -44,6 +55,101 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+@for_all_methods(
+    retries,
+    stop=stop_after_attempt(
+        int(os.environ.get("MONGO_AUTO_RETRY_STOP_AFTER_ATTEMPT", 3))
+    ),
+    wait=wait_random(
+        min=int(os.environ.get("MONGO_AUTO_RETRY_WAIT_RANDOM_MIN", 0.1)),
+        max=int(os.environ.get("MONGO_AUTO_RETRY_WAIT_RANDOM_MAX", 1)),
+    ),
+    before_sleep=before_sleep,
+    retry=retry_if_exception_type((AutoReconnect,)),
+)
+class UserController:
+    """UserController"""
+
+    # pylint: disable=unnecessary-lambda
+    def __init__(self, get_mongo=lambda: Mongo()):
+        self.mongo = get_mongo()
+        self.db_client = self.mongo.client
+        self.db = self.db_client[self.mongo.db_name]
+
+    def bootstrap_indexes(self):
+        """Bootstrap all users related indexes."""
+        index_tuples = [
+            ("users", [("username", pymongo.ASCENDING)], {"unique": True}),
+        ]
+        for collection, keys, kwargs in index_tuples:
+            if self.mongo.bootstrap_index(collection, keys, **kwargs):
+                LOG.info(
+                    f"Created DB index {keys}, collection: {collection}"
+                )
+
+    def create_user(self, user_data) -> pymongo.InsertOne:
+        """Create user to database"""
+        try:
+            utc_now = datetime.datetime.utcnow()
+            user_data.update({"inserted_at": utc_now})
+            user_data.update({"updated_at": utc_now})
+            result = self.db.users.insert_one(UserDoc(**user_data).dict())
+        except (ValidationError, DuplicateKeyError) as e:
+            raise e
+        return result
+
+    def delete_user(self, username) -> dict:
+        """Delete user from database"""
+        utc_now = datetime.datetime.utcnow()
+        update = {
+                "state": "inactive",
+                "deleted_at": utc_now,
+        }
+        result = self.db.users.find_one_and_update(
+            {"username": username},
+            {"$set": update},
+            return_document=ReturnDocument.AFTER,
+        )
+        return result
+
+    def update_user(self, username, data) -> dict:
+        """Update user from database"""
+        try:
+            data.update({"updated_at": datetime.datetime.utcnow()})
+            validated_data = UserDocUpdate(**data).dict(exclude_none=True)
+        except ValidationError as e:
+            raise e
+        result = self.db.users.find_one_and_update(
+            {"username": username},
+            {"$set": validated_data},
+            return_document=ReturnDocument.AFTER
+        )
+        return result
+
+    def get_user(self, username) -> dict:
+        """Return a user information from database"""
+        data = self.db.users.aggregate([
+            {"$match": {"username": username}},
```

Comment:
`username` is indexed and unique, so we only expect a single result, let's make the query clearer and explicit with `{"$limit": 1}` in this aggregation pipeline, just so it's immediate evident for other developers who will also read and maintain this code.

Reply:
Got it.